lib/roken/parse_bytes.c

```diff
@@ -33,10 +33,27 @@
 
 #include <config.h>
 
+#include <stdio.h>
 #include <parse_units.h>
 #include "parse_bytes.h"
 
+/*
+ * We need a way to express units that might be too large to be possible to
+ * represent in ssize_t.
+ *
+ * For example, if sizeof(ssize_t) == 4 then we can't represent terabytes, but
+ * we should still be able to have declare entry for terabytes here and just
+ * not use it.
+ *
+ * The simplest way to do this is to skip as many entries as needed to reach a
+ * strictly-decreasing set of entries using casts to size_t below to get
+ * unsigned overflow semantics.  This allows us to have entries for
+ * terabyte which, if sizeof(ssize_t) == 4, get skipped over.
+ */
 static struct units bytes_units[] = {
+    { "terabyte", (size_t)1024 * 1024 * 1024 * 1024 },
```

Comment:
Actually, the second field is a `uint64_t` now, so there's no need to do this cast or check for overflow.

Reply:
I would prefer this as `0x10000000000LL` instead of the multiplication.